eva_submission/submission_qc_checks.py

```diff
@@ -0,0 +1,316 @@
+                            return True
+        return False
+
+    def get_species_name(self, assembly):
+        with get_metadata_connection_handle(self.profile, self.private_config_xml_file) as pg_conn:
+            query = f"""select concat(t.taxonomy_code, '_',a.assembly_code) from evapro.taxonomy t 
+                        join evapro.assembly a on a.taxonomy_id = t.taxonomy_id 
+                        where t.taxonomy_id = {self.taxonomy} and assembly_accession='{assembly}'"""
+            return get_all_results_for_query(pg_conn, query)[0][0]
+
+    def get_browsable_files_for_study(self):
+        with get_metadata_connection_handle(self.profile, self.private_config_xml_file) as pg_conn:
+            query = f"select filename from evapro.browsable_file where project_accession='{self.project_accession}'"
+            return [filename for filename, in get_all_results_for_query(pg_conn, query)]
+
+    def check_all_browsable_files_are_available_in_ftp(self, vcf_files):
+        try:
+            files_in_ftp = self.get_files_from_ftp(self.project_accession)
+        except Exception as e:
+            logger.error(f"Error fetching files from ftp for study {self.study_accession}. Exception  {e}")
+            self._ftp_check_result = "FAIL"
+            return f"""
+                Error: Error fetching files from ftp for study {self.project_accession}"""
+
+        if not files_in_ftp:
+            logger.error(f"No file found in ftp for study {self.study_accession}")
+            self._ftp_check_result = "FAIL"
+            return f"""
+                Error: No files found in FTP for study {self.project_accession}"""
+
+        missing_files = []
+
+        for file in vcf_files:
+            if file not in files_in_ftp:
+                missing_files.append(file)
+            if f'{file}.csi' not in files_in_ftp:
+                missing_files.append(f'{file}.csi')
+
+            # accessioned files will not be present for human taxonomy
+            if self.taxonomy != 9606:
+                accessioned_file = file.replace('.vcf.gz', '.accessioned.vcf.gz')
+                if accessioned_file not in files_in_ftp:
+                    missing_files.append(accessioned_file)
+                if f'{accessioned_file}.csi' not in files_in_ftp:
+                    missing_files.append(f'{accessioned_file}.csi')
+
+        self._ftp_check_result = "PASS" if not missing_files else "FAIL"
+        return f"""
+                pass: {self._ftp_check_result} 
+                missing files: {missing_files if missing_files else None}"""
+
+    @retry(tries=3, delay=2, backoff=1.5, jitter=(1, 3))
+    def get_files_from_ftp(self, project_accession):
+        ftp = FTP('ftp.ebi.ac.uk', timeout=600)
+        ftp.login()
+        ftp.cwd(f'pub/databases/eva/{project_accession}')
+        return ftp.nlst()
+
+    def check_if_job_completed_successfully(self, file_path):
+        with open(file_path, 'r') as f:
```

Comment:
```suggestion
        with open(file_path, 'r') as f:
            job_status = 'FAILED'
```
Need this in case the process crashes before the job is launched, otherwise the `job_status` variable is undefined.  And similar in the other methods where we scan through the log file.